File: db4on/core/src/core/com/db4o/cs/messages/MTaDelete.cs
namespace com.db4o.cs.messages
{
	public class MTaDelete : com.db4o.cs.messages.MsgD
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 @in)
		{
			int id = _payLoad.ReadInt();
			int cascade = _payLoad.ReadInt();
			com.db4o.Transaction trans = GetTransaction();
			com.db4o.YapStream stream = trans.Stream();
			lock (stream.i_lock)
			{
				object[] arr = stream.GetObjectAndYapObjectByID(trans, id);
				trans.Delete((com.db4o.YapObject)arr[1], cascade);
				return true;
			}
		}
	}
}

File: db4oj/core/src/com/db4o/internal/cs/messages/MTaDelete.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.cs.messages;

import com.db4o.internal.*;

public class MTaDelete extends MsgD implements ServerSideMessage {
	
	public final boolean processAtServer() {
	    int id = _payLoad.readInt();
	    int cascade = _payLoad.readInt();
	    Transaction trans = transaction();
	    synchronized (streamLock()) {
	        trans.delete(null, id, cascade);
	        return true;
	    }
	}
}
